src/wast-parser.cc

```diff
@@ -3100,10 +3165,9 @@ Result WastParser::ParseBlockDeclaration(BlockDeclaration* decl) {
   WABT_TRACE(ParseBlockDeclaration);
   FuncDeclaration func_decl;
   CHECK_RESULT(ParseTypeUseOpt(&func_decl));
-  CHECK_RESULT(ParseUnboundFuncSignature(&func_decl.sig));
+  CHECK_RESULT(ParseUnboundFuncSignature(&decl->sig));
```

Comment:
Using locals for storing type vector is not possible anymore. I hope this is the only part where it caused an issue. I cannot check it with asserts.

This is the weakness of the patch.